core/src/main/java/com/google/instrumentation/tags/TagKey.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.tags;
+
+import static com.google.instrumentation.tags.TagKey.TagType.TAG_BOOL;
+import static com.google.instrumentation.tags.TagKey.TagType.TAG_INT;
+import static com.google.instrumentation.tags.TagKey.TagType.TAG_STRING;
+
+import com.google.auto.value.AutoValue;
+import com.google.instrumentation.internal.StringUtil;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Tag key.
```

Comment:
I wasn't sure what to write, but I tried to add an explanation in the TagSet Javadoc.  Are the new comments accurate?